hydroflow/tests/surface_scheduling.rs

```diff
@@ -121,3 +123,43 @@ pub fn test_issue_800_1050_fold_keyed() {
     assert_graphvis_snapshots!(df);
     df.run_available();
 }
+
+#[multiplatform_test(hydroflow, env_tracing)]
+async fn test_nospin_issue_961() {
+    let mut df = hydroflow_syntax! {
+        source_iter([1])
+            -> next_stratum()
+            -> persist()
+            -> defer_tick_lazy()
+            -> null();
+    };
+    assert_graphvis_snapshots!(df);
+
+    timeout(Duration::from_millis(100), df.run_available_async())
```

Comment:
Isn't it sufficient to just run df.run_available_async() and check if it returns? If this test fails, it runs forever and some top-level timeout can catch that? Perhaps configured at a github runner level? 

I'm unclear if there are false positives with a choice like 100ms - perpetually.

Reply:
Yeah, we also may want to raise the timeout if there's no way to deterministically test this. Running tests with `nextest` does tend to stretch out latencies a bit.

Reply:
The timeout is high enough (code should take about 0.5ms to run), but would be nice to have a cleaner way to test